### DISCUSSION TITLE
Everyday Usage

### DISCUSSION THREAD

**Discussion by osalbahr** [Homebrew/discussions#4715] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.4-38-g7044f50
ORIGIN: https://github.com/Homebrew/brew
HEAD: 7044f50f889f8ec7d8656ac176e960e53bee1a0a
Last commit: 14 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: fd6b0c5015dafc1c9e18943af0147d04736e7d66
Core tap last commit: 6 days ago
Core tap branch: master
Core tap JSON: 13 Aug 01:28 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit kabylake
Clang: 14.0.3 build 1403
Git: 2.41.0 => /usr/local/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 13.5-x86_64
CLT: 14.3.1.0.1.1683849156
Xcode: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I noticed that I have [temurin](https://formulae.brew.sh/cask/temurin) installed but I wasn't sure if I directly installed it or if it was installed as a dependency. However `brew uses` tells me that there is no available formula. I also tried with `--cask`. Is this the expected behavior?

```console
% brew uses --installed temurin
Warning: No available formula with the name "temurin". Did you mean maturin?
% brew uses --installed --cask temurin
Warning: No available formula with the name "temurin". Did you mean maturin?
% brew install temurin
Warning: Not upgrading temurin, the latest version is already installed
```

I tried `brew bundle dump` but I'm not sure if it thinks I installed it on-demand because I specifically did `brew install temurin` above (to double-check that it's already installed).

```console
% brew bundle dump --file=/dev/stdout | grep temurin
cask "temurin"
```

---

Edit: I just realized `brew uses` works only for formulae. Can a Cask be a dependency?

**Comment by apainintheneck** [comment]
Casks are not allowed as dependencies for formula (at least in `homebrew/core`) because we want to be able to build everything from source for bottling purposes. Casks can depend on other casks though (https://docs.brew.sh/Cask-Cookbook#stanza-depends_on).

It doesn't look like this package is used as a dependency in `homebrew/cask`.

```sh
/u/l/H/L/T/h/homebrew-cask (master|✔) $ git grep 'depends_on cask: "temurin"'
/u/l/H/L/T/h/homebrew-cask (master|✔) [1]$
```

If a package is first installed as a dependency but then you run `brew install pkg_name` for it, we will update the install info to reflect that fact (at least for formula; I'll have to double-check about casks).
